tests/by-util/test_cksum.rs

```diff
@@ -805,4 +805,97 @@ fn test_cksum_check_failed() {
         .stderr_str()
         .contains("1 listed file could not be read\n"));
     assert!(result.stdout_str().contains("f: OK\n"));
+
+    // tests with two files
+    at.touch("CHECKSUM2");
+    at.write("f2", "42");
+    for command in &commands {
+        let result = scene.ucmd().args(command).arg("f2").succeeds();
+        at.append("CHECKSUM2", result.stdout_str());
+    }
+    // inject invalid content
+    at.append("CHECKSUM2", "again incorrect data\naze\nSM3 (input2) = 7cfb120d4fabea2a904948538a438fdb57c725157cb40b5aee8d937b8351477e\n");
+    at.append("CHECKSUM2", "again incorrect data\naze\nSM3 (input2) = 7cfb120d4fabea2a904948538a438fdb57c725157cb40b5aee8d937b8351477e\n");
+
+    let result = scene
+        .ucmd()
+        .arg("--check")
+        .arg("CHECKSUM")
+        .arg("CHECKSUM2")
+        .fails();
+    println!("result.stderr_str() {}", result.stderr_str());
+    println!("result.stdout_str() {}", result.stdout_str());
+    assert!(result.stderr_str().contains("Failed to open file: input2"));
+    assert!(result
+        .stderr_str()
+        .contains("4 lines are improperly formatted\n"));
+    assert!(result
+        .stderr_str()
+        .contains("2 listed files could not be read\n"));
+    assert!(result.stdout_str().contains("f: OK\n"));
+    assert!(result.stdout_str().contains("2: OK\n"));
+}
+
+#[test]
+fn test_check_md5_format() {
+    let scene = TestScenario::new(util_name!());
+
+    let at = &scene.fixtures;
+    at.touch("empty");
+    at.write("f", "d41d8cd98f00b204e9800998ecf8427e *empty\n");
+
+    scene
+        .ucmd()
+        .arg("-a")
+        .arg("md5")
+        .arg("--check")
+        .arg("f")
+        .succeeds()
+        .stdout_contains("empty: OK");
+
+    // with a second file
+    at.write("not-empty", "42");
+    at.write("f2", "a1d0c6e83f027327d8461063f4ac58a6 *not-empty\n");
+
+    scene
+        .ucmd()
+        .arg("-a")
+        .arg("md5")
+        .arg("--check")
+        .arg("f")
+        .arg("f2")
+        .succeeds()
+        .stdout_contains("empty: OK")
+        .stdout_contains("not-empty: OK");
+}
+
+// Manage the mixed behavior
+// cksum --check -a sm3 CHECKSUMS
+// when CHECKSUMS contains:
```

Comment:
This comment is misleading.
```suggestion
// when CHECKSUM contains among other lines:
```
Also, may I suggest again that we avoid creating "expected data" on-the-fly from the exact same code that is currently under test?